### PR TITLE
chore: reorganize and restructure code

### PR DESCRIPTION
**Pull request by mkarajohn**

### CODE REVIEW

**Comment by mkarajohn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #101** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/101" target="_blank">Graphite</a> 
    * **PR #103** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/103" target="_blank">Graphite</a> 
      * **PR #105** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/105" target="_blank">Graphite</a> 
        * **PR #102** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/102" target="_blank">Graphite</a> 
          * **PR #106** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/106" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/106?utm_source=stack-comment).

File: src/providers/Organizations.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { ReactNode, useCallback, useMemo } from 'react';
+import { OrganizationsContext } from '../contexts/organizations';
+import useOrganization from '../store/organizations';
+import { getAuth0Client } from '../utils/auth';
+
+export function Organizations(props: { children: ReactNode }) {
```
Comment (panvourtsis) [inline src/providers/Organizations.tsx:6]: is this context only to apply the switching? Otherwise all other functions are coming from the store

Comment (mkarajohn) [reply src/providers/Organizations.tsx:6]: Yes + the whole idea is that we are only exposing *specific* functionality through React hooks and keep implementation details such as the fact that we use zustand hidden from the end user. What is exposed should be enough. If it's not we update the API accordingly.

Comment (panvourtsis) [reply src/providers/Organizations.tsx:6]: i don't see any benefits in wrapping things in a new context but I see it as more boilerplate. Exposing zustand state as app hook fulfills all cases. Is there something else I might miss?

Comment (mkarajohn) [reply src/providers/Organizations.tsx:6]: Wrapping it in context and passing it down by context
* makes a clear the separation of different kinds of data
* makes the flow of information 100% clear among the contexts, you can see which context needs which context before it's ready 
* most importantly, abstracts away the implementation detail that is zustand. If tomorrow we replace it with something else all we need to change are the internals of this context provier.

It's not boilerplate, it's done so in order to follow a specific pattern of logic

**Review by panvourtsis — APPROVED** [review]

**Comment by mkarajohn** [comment]
### Merge activity

* **Mar 13, 7:45 AM EDT**: @mkarajohn started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/106).

**Pull request merged**